src/client/pydaos/pydaos_core.py

```diff
@@ -69,7 +69,7 @@ class DCont():
         Create new DArray object.
     """
 
-    def __init__(self, pool=None, cont=None, path=None):
+    def __init__(self, pool=None, cont=None, path=None, open_ro=False):
```

Comment:
Instead of `open_ro=True` or `open_ro=False`, should we just take a `flags` argument? Or, for python convenience, we could even accept a string of `RW` or `RO`.
`open_ro` seems oddly specific

Reply:
It is oddly specific but I was trying to maintain 100% compatibility with existing code where no mode is provided.  Yes, in hindsight the interface should have required a `RW` or `RO` option from the start.

Perhaps this would be better?  Then we could support `RW` or `RO`.
```suggestion
    def __init__(self, pool=None, cont=None, path=None, open_mode='RW`):
```

Reply:
I think that's good. And defaulting to `RW` is backwards compatible

Reply:
The current PR is backwards compatible, that's why I wrote it how I did.  I can update tomorrow.

Reply:
The current is backward compatible, but if we ever need to open with a different flag in the future, we'll need to add another argument. Whereas a generic `flags` argument makes it easier for future changes. E.g. if someone wanted to use `DAOS_COO_EX`.

Also, defaulting `flags=RW` makes it clear that the default is `RW`. But `open_ro=False` only tells you that it's NOT `RO`.